### PR TITLE
Fixed multiple VS compiler warning

### PR DESCRIPTION
**Pull request by MrKeiKun**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

**Issues addressed:** 

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by MrKeiKun** [comment]
_Ignore the branch name_

File: src/map/clif.h [hunk 1]
```diff
@@ -1039,8 +1039,8 @@ struct clif_interface {
 	void (*initialstatus) (struct map_session_data *sd);
 	void (*cooldown_list) (int fd, struct skill_cd* cd);
 	/* player-unit-specific-related */
-	void (*updatestatus) (struct map_session_data *sd,int type);
-	void (*changestatus) (struct map_session_data* sd,int type,int val);
+	void (*updatestatus) (struct map_session_data *sd, enum type);
```
Comment (Asheraf) [inline src/map/clif.h:1042]: `enum status_point_types`

**Review by Asheraf — CHANGES_REQUESTED** [review]

File: src/map/clif.h (hunk 1 above)
Comment (MrKeiKun) [reply src/map/clif.h:1042]: holy crap. nice catch!

File: src/map/clif.h [hunk 2]
```diff
@@ -1039,8 +1039,8 @@ struct clif_interface {
 	void (*initialstatus) (struct map_session_data *sd);
 	void (*cooldown_list) (int fd, struct skill_cd* cd);
 	/* player-unit-specific-related */
-	void (*updatestatus) (struct map_session_data *sd,int type);
-	void (*changestatus) (struct map_session_data* sd,int type,int val);
+	void (*updatestatus) (struct map_session_data *sd, enum status_point_types);
```
[the remark this answers is not in the document]
Comment (MrKeiKun) [reply src/map/clif.h]: have to replace the ctrl key after this, stuck on multi selection :|

File: src/map/clif.c [hunk 3]
```diff
@@ -4566,7 +4566,8 @@ static void clif_tradeadditem(struct map_session_data *sd, struct map_session_da
 		else
 			p.itemId = sd->status.inventory[index].nameid;
 #if PACKETVER >= 20100223
-		p.itemType = sd->inventory_data[index]->type;
+		if (sd->inventory_data[index] != NULL)
```
Comment (Asheraf) [inline src/map/clif.c:4569]: this is already being checked few lines above, you can probably replace both checks with a nullpo_retv under the assert.

**Review by 4144 — APPROVED** [review]

**Review by MishimaHaruna — APPROVED** [review]

**Pull request merged**